src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -164,6 +164,8 @@ internal sealed override bool GenerateDebugInfo
             }
         }
 
+        public sealed override bool AreLocalsZeroed => ContainingType.AreLocalsZeroed;
```

Comment:
>public sealed override bool AreLocalsZeroed => ContainingType.AreLocalsZeroed; [](start = 8, length = 78)

It looks like the base implementation should be able to return the right value. Probably implementation of ```AreContainingSymbolLocalsZeroed``` should be adjusted instead.

Reply:
You're right. I got too caught up in the fact that a lot of types (including [`SimpleProgramNamedTypeSymbol`](https://github.com/dotnet/roslyn/blob/8861cf5e10d5d462acaa85020af33d6a2f6885cd/src/Compilers/CSharp/Portable/Symbols/Source/SimpleProgramNamedTypeSymbol.cs#L128-L131)) override `AreLocalsZeroed` that I forgot to check that.

I believe the real issue is here:

https://github.com/dotnet/roslyn/blob/ac33391b666ee25e3e977d6c14f3fdf2e21d20db/src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs#L47-L50

`SourceNamedTypeSymbol` seems overly specific since `ContainingType` is a `NamedTypeSymbol` which already has `AreLocalsZeroed` directly. Changing this to be a null check also fixes the issue, so I've change the fix to prefer that instead.

Reply:
@RikkiGibson You introduced this logic in https://github.com/dotnet/roslyn/pull/41183

Do you remember if there was a reason you constrained `ContainingType` to `SourceNamedTypeSymbol `?

Reply:
No, I think it would be fine for it to be a NamedTypeSymbol instead.

Reply:
Alright, thanks for the confirmation. I locally ran the emit tests with `NamedTypeSymbol` and they all passed. So we'll see if the other tests reveal any unintended consequences of going that route once CI is finished.

Reply:
Looks like this causes us to fail EE tests by hitting an assert in `PENamedTypeSymbol.AreLocalsZeroed.get`. So changing it to `SourceMemberContainerTypeSymbol` is probably the easiest fix.

Reply:
Yup. I should've maybe anticipated that one considering I saw the `AreLocalsZeroed` implementations that threw. Changed it to `SourceMemberContainerTypeSymbol` and those tests pass locally now.